crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -37,6 +39,57 @@ impl DynamicScene {
         Self::from_world(&scene.world, type_registry)
     }
 
+    /// Create a Dynamic Scene with specific entities.
+    ///
+    /// The generic parameter is used as a `Query` filter.
+    ///
+    /// The created scene will include only the entities that match the query
+    /// filter provided. All components that impl `Reflect` will be included.
+    pub fn from_query_filter<F>(world: &mut World) -> Self
+    where
+        F: ReadOnlyWorldQuery + 'static,
+    {
+        let mut ss = SystemState::<Query<Entity, F>>::new(world);
```

Comment:
We can probably get away with `world.query` here, which has reduced overhead.

Reply:
Done!